igvc/src/motor_controller/main.cpp

```diff
@@ -28,66 +35,164 @@ int main(int argc, char** argv)
-  {
-    ros::spinOnce();
 
-    std::string msg = "$" + std::to_string(enabled ? current_motor_command.left_velocity : 0.0) + "," +
-                      std::to_string(enabled ? current_motor_command.right_velocity : 0.0) + "\n";
+  std::string p_values = "#P" + toBoundedString(p_l) + "," + toBoundedString(p_r) + "\n";
+  std::string d_values = "#D" + toBoundedString(d_l) + "," + toBoundedString(d_r) + "\n";
+  std::string i_values = "#I" + toBoundedString(i_l) + "," + toBoundedString(i_r) + "\n";
 
-    if (fabs(current_motor_command.left_velocity) > 0 || fabs(current_motor_command.right_velocity) > 0)
-      ROS_ERROR_STREAM("msg to motor arduino $lv,rv\n" << msg);
+  bool valid_values_p = false;
+  bool valid_values_d = false;
+  bool valid_values_i = false;
+
+  while(ros::ok() && (!valid_values_p || !valid_values_d || !valid_values_i)) {
+    ros::spinOnce();
+    ROS_INFO_STREAM("sending PID values to board");
+    if(!valid_values_p) {
+      port.write(p_values);
+    }
+    if(!valid_values_d) {
+      port.write(d_values);
+    }
+    if(!valid_values_i) {
+      port.write(i_values);
+    }
+    for(int i = 0; i < 3; i++) {
+      std::string ret = port.readln();
+      if(!ret.empty() && ret.at(0) == '#' && ret.at(1) != 'E') {
+        size_t p_loc = ret.find('P');
+        size_t d_loc = ret.find('D');
+        size_t i_loc = ret.find('I');
+        size_t end = ret.find('\n');
+        if(p_loc != std::string::npos) {
+          std::vector<std::string> vals = split(ret.substr(p_loc + 1, end), ',');
+          ROS_INFO_STREAM("p ret = " << ret);
+          if(vals.size() == 2) {
+            valid_values_p = true;
+            valid_values_p = (stod(vals.at(0)) == p_l) && valid_values_p;
+            valid_values_p = (stod(vals.at(1)) == p_r) && valid_values_p;
+          }
+        } else if(d_loc != std::string::npos) {
+          std::vector<std::string> vals = split(ret.substr(d_loc + 1, end), ',');
+          ROS_INFO_STREAM("d ret = " << ret);
+          if(vals.size() == 2) {
+            valid_values_d = true;
+            valid_values_d = (stod(vals.at(0)) == d_l) && valid_values_d;
+            valid_values_d = (stod(vals.at(1)) == d_r) && valid_values_d;
+          }
+        } else if(i_loc != std::string::npos) {
+          std::vector<std::string> vals = split(ret.substr(i_loc + 1, end), ',');
+          ROS_INFO_STREAM("i ret = " << ret);
+          if(vals.size() == 2) {
+            valid_values_i = true;
+            valid_values_i = (stod(vals.at(0)) == i_l) && valid_values_i;
+            valid_values_i = (stod(vals.at(1)) == i_r) && valid_values_i;
+          }
+        } else {
+          ROS_INFO_STREAM("recieved unknown string while setting PID values " << ret);
+        }
+      } else if(ret.empty()) {
+        ROS_ERROR_STREAM("Empty return from motor arduino while sending PID values.\t" << ret);
+      }
+    }
+    rate.sleep();
+  }
+  ROS_INFO_STREAM("sucessfully sent PID values");
```

Comment:
Sometimes the first letter of print statements are capitalized, but other time's it's not.  This is something that would ideally be consistent once we're out of the debugging stage.